Tools/LambdaTestTool/src/Amazon.Lambda.TestTool.BlazorTester/Services/RuntimeApiDataStore.cs

```diff
@@ -0,0 +1,322 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading;
+
+namespace Amazon.Lambda.TestTool.BlazorTester.Services
+{
+    public interface IRuntimeApiDataStore
+    {
+        EventContainer QueueEvent(string eventBody);
+        
+        IReadOnlyList<IEventContainer> QueuedEvents { get; }
+        
+        IReadOnlyList<IEventContainer> ExecutedEvents { get; }
+
+        void ClearQueued();
+
+        void ClearExecuted();
+
+        void DeleteEvent(string awsRequestId);
+
+
+        IEventContainer ActiveEvent { get; }
+
+        event EventHandler StateChange;
+
+        bool TryActivateEvent(out IEventContainer activeEvent);
+
+        void ReportSuccess(string awsRequestId, string response);
+        void ReportError(string awsRequestId, string errorType, string errorBody);
+    }
+
+    public class RuntimeApiDataStore : IRuntimeApiDataStore
+    {
+        private IList<EventContainer> _queuedEvents = new List<EventContainer>();
+        private IList<EventContainer> _executedEvents = new List<EventContainer>();
+        private int _eventCounter = 1;
+        private object _lock = new object();
+        
+        public event EventHandler StateChange;
+        
+        public EventContainer QueueEvent(string eventBody)
+        {
+            var evnt = new EventContainer(this, _eventCounter++, eventBody);
+            Monitor.Enter(_lock);
```

Comment:
Curious on knowing usage of `Monitor` over `lock`. 

I always though Monitor is low level construct while `lock` is abstraction which handles the cirtical section release based on the code block which is better as developer don't have to worry about release/exit.
```csharp
lock(_obc)
{
}
```

Reply:
Switched to using `lock`. I believe an earlier iteration of this class involved async operations used within the lock so I couldn't use `lock` in that case.